accounting/filters/OsgScheddCpuRemovedFilter.py

```diff
@@ -0,0 +1,400 @@
+        self.collector_host = "flock.opensciencegrid.org"
+        self.schedd_collector_host_map = {}
+        super().__init__(**kwargs)
+
+    def schedd_collector_host(self, schedd):
+        # Query Schedd ad in Collector for its CollectorHost,
+        # unless result previously cached
+        if schedd not in self.schedd_collector_host_map:
+
+            collector = htcondor.Collector(self.collector_host)
+            ads = collector.query(
+                htcondor.AdTypes.Schedd,
+                constraint=f'''Machine == "{schedd.split('@')[-1]}"''',
+                projection=["CollectorHost"],
+            )
+            ads = list(ads)
+            if len(ads) == 0:
+                logging.warning(f'Could not find Schedd ClassAd for Machine == "{schedd}"')
+                logging.warning(f"Assuming jobs from {schedd} are in OS pool")
+                self.schedd_collector_host_map[schedd] = self.collector_host
+                return self.collector_host
+            if len(ads) > 1:
+                logging.warning(f'Got multiple Schedd ClassAds for Machine == "{schedd}"')
+
+            # Cache the CollectorHost in the map
+            if "CollectorHost" in ads[0]:
+                self.schedd_collector_host_map[schedd] = ads[0]["CollectorHost"].split(':')[0]
+            else:
+                logging.warning(f"CollectorHost not found in Schedd ClassAd for {schedd}")
+                self.schedd_collector_host_map[schedd] = "UNKNOWN"
+
+        return self.schedd_collector_host_map[schedd]
+
+    def schedd_filter(self, data, doc):
+
+        # Get input dict
+        i = doc["_source"]
+
+        # Get output dict for this schedd
+        schedd = i.get("ScheddName", "UNKNOWN") or "UNKNOWN"
+        o = data["Schedds"][schedd]
+
+        # Filter out jobs that were not removed
+        if i.get("JobStatus",4) != 3:
+            return
+
+        # Filter out jobs that did not run in the OS pool        
+        if i.get("LastRemotePool", self.schedd_collector_host(schedd)) != self.collector_host:
+            return
+
+        # Get list of attrs
+        filter_attrs = DEFAULT_FILTER_ATTRS.copy()
+
+        # Count number of history ads (i.e. number of unique job ids)
+        o["_NumJobs"].append(1)
+
+        # Do filtering for scheduler and local universe jobs
+        univ = i.get("JobUniverse", 5)
+        o["_NumSchedulerUnivJobs"].append(univ == 7)
+        o["_NumLocalUnivJobs"].append(univ == 12)
+        o["_NoShadow"].append(univ in [7, 12])
+
+        # Compute badput fields
+        if (
+                univ not in [7, 12] and
+                i.get("NumJobStarts", 0) > 1 and
+                i.get("RemoteWallClockTime", 0) > 0 and
+                i.get("RemoteWallClockTime") != i.get("CommittedTime")
+            ):
+            o["_BadWallClockTime"].append(i["RemoteWallClockTime"] - i.get("CommittedTime", 0))
+            o["_NumBadJobStarts"].append(i["NumJobStarts"] - 1)
```

Comment:
All `RemoteWallClockTime` and `NumJobStarts` are "badput" here, unless `CommittedTime > 0`.
```suggestion
        if (
                univ not in [7, 12] and
                i.get("NumJobStarts", 0) > 0 and
                i.get("RemoteWallClockTime", 0) > 0 and
                i.get("RemoteWallClockTime") != i.get("CommittedTime")
            ):
            if i.get("CommittedTime", 0) > 0:
                o["_BadWallClockTime"].append(i["RemoteWallClockTime"] - i.get("CommittedTime", 0))
                o["_NumBadJobStarts"].append(i["NumJobStarts"] - 1)
            else:
                o["_BadWallClockTime"].append(i["RemoteWallClockTime"])
                o["_NumBadJobStarts"].append(i["NumJobStarts"])
```